### PR TITLE
7.2

### PR DESCRIPTION
**Pull request by extremeshok**
### Version 7.2 (07 December 2020)

* Database rating downgrades are now supported, eg, changing from HIGH to LOW will remove the HIGH and MEDIUM rated databases.
* Disabled databases are automatically removed
* Disable databases by  setting the rating to "DISABLED" eg. securiteinfo_dbs_rating="DISABLED" will disable all securiteinfo databases
* Added Malware Expert databases (non-free)
* Added interServer databases (free)
* Reworked securiteinfo premium databases (non-free)
* Added malwarepatrol_db to specify the exact database name (default: malwarepatrol.db)
* Added detection of tar executable (use gtar on mac and bsd)
* Config os.macosx.conf renamed to os.macos.conf
* Fix: set ownership of last-version-check.txt
* More automated linting and testing (markdown and macOS / osx) via travis-ci
* Updated macOS installation guide for Big Sur (OSX 11)
* Incremented the config to version 94
* Thank you @dandanio @jkellerer @msapiro @shawniverson

### Version 7.1 (Not Released)

* Enforce HTTPS validation by default
* Updated sanesecurity publickey.gpg url to use SSL
* Ignore yara files that include modules
* Enabled yararulesproject rules by default
* os.gentoo.conf: disable updates and upgrade checks
* Fix: URLhaus log message
* Fix wrong download URL for MalwarePatrol
* Fix: fallback to host if dig is not used
* Disable cron MAILTO
* BSD read config fix
* Incremented the config to version 92
* Thank you @dandanio @jkellerer @m0urs @Mrothyr @msapiro @orlitzky @RobbieTheK @SlothOfAnarchy

### CODE REVIEW

**Comment by extremeshok** [comment]
Would be appreciated if any of you are able to check/test
@dandanio @Mrothyr @m0urs @msapiro @orlitzky @SlothOfAnarchy @jkellerer @RobbieTheK

**Comment by SlothOfAnarchy** [comment]
Config parsing fails on Linux (NixOS) for me, I think because of https://github.com/extremeshok/clamav-unofficial-sigs/commit/dd48c01854788f5d3279756fe726d080e70f9ed2?branch=dd48c01854788f5d3279756fe726d080e70f9ed2&diff=unified

File: clamav-unofficial-sigs.sh [hunk 1]
```diff
@@ -2504,23 +2528,30 @@ if [ "$sanesecurity_enabled" == "yes" ] ; then
         echo "$current_time" > "${work_dir_work_configs}/last-ss-update.txt"
         xshok_pretty_echo_and_log "Sanesecurity Database & GPG Signature File Updates" "="
         xshok_pretty_echo_and_log "Checking for Sanesecurity updates..."
-				# shellcheck disable=SC2086
-        sanesecurity_mirror_ips="$(dig $dig_proxy +ignore +short "$sanesecurity_url")"
-        # Add fallback to host if dig returns no records
+				if [ -z "$dig_bin" ] ; then
```
Comment (jkellerer) [inline clamav-unofficial-sigs.sh:2531]: This condition is inverted. It should be `if [ ! -z "$dig_bin" ] ; then` (if dig exists, then use it)

File: clamav-unofficial-sigs.sh [hunk 2]
```diff
@@ -2504,23 +2528,30 @@ if [ "$sanesecurity_enabled" == "yes" ] ; then
...
+          sanesecurity_mirror_ips="$($host_bin $host_proxy -t A "$sanesecurity_url" | sed -n '/has address/{s/.*address \([^ ]*\).*/\1/;p;}')"
         fi
 
         if [ ${#sanesecurity_mirror_ips} -ge 1 ] ; then
           for sanesecurity_mirror_ip in $sanesecurity_mirror_ips ; do
-            sanesecurity_mirror_name=""
-						# shellcheck disable=SC2086
-            sanesecurity_mirror_name="$(dig $dig_proxy +short -x "$sanesecurity_mirror_ip" | command sed 's/\.$//')"
-            # Add fallback to host if dig returns no records
+						if [ -z "$dig_bin" ] ; then
```
Comment (jkellerer) [inline clamav-unofficial-sigs.sh:2545]: Same issue as line 2531, inverted condition.

**Comment by msapiro** [comment]
I have two issues. The first is that I have `/usr/local/sbin/clamav-unofficial-sigs` which is a symlink to `clamav-unofficial-sigs.sh`. If I run `/usr/local/sbin/clamav-unofficial-sigs` I get `ERROR: could not determin script name and fullpath` because at line 1509
```
  this_script_full_path="$(readlink "$this_script_full_path")"
```
sets `this_script_full_path` to `clamav-unofficial-sigs.sh` and then line 1516
```
this_script_path="$( cd -P "$( dirname "$this_script_full_path" )" >/dev/null 2>&1 && readlink -f "$0" )"
```
sets `this_script_name` null.

If I run `/usr/local/sbin/clamav-unofficial-sigs.sh`, I get past that but get
```
################################################################################
 eXtremeSHOK.com ClamAV Unofficial Signature Updater
 Version: v7.0.1 (2020-01-25)
 Required Configuration Version: v92
 Copyright (c) Adrian Jon Kriel :: admin@extremeshok.com
################################################################################
Loading config: /etc/clamav-unofficial-sigs/master.conf
===================================================================
ERROR: Your configuration has errors, every = requires a pair of ""
===================================================================
```
I'm puzzled bu this one because if I do
```
clean_config="$(command sed -e '/^#./d' -e 's/[[:space:]]#.//' -e 's/#[[:space:]].//' -e $'s/^[ \t]//;s/[ \t]$//' -e '/^\s$/d' /etc/clamav-unofficial-sigs/master.conf)"
clean_config="${clean_config//|/\\|}"
config_check="${clean_config//[^\"]}"
echo ${#config_check}
```
the result is `160`

**Comment by extremeshok** [comment]
> I have two issues. The first is that I have `/usr/local/sbin/clamav-unofficial-sigs` which is a symlink to `clamav-unofficial-sigs.sh`. If I run `/usr/local/sbin/clamav-unofficial-sigs` I get `ERROR: could not determin script name and fullpath` because at line 1509

$PWD is failing under sudo, busy testing a workaround

**Comment by extremeshok** [comment]
ready for 7.2 release ?

**Comment by msapiro** [comment]
This latest solves my earlier issues and runs OK. I need to work on my config as I'm not getting some sigs I got previously, but that's probably just a config issue.

**Comment by shawniverson** [comment]
@extremeshok I can publish this to my testing repos for CentOS 7 and 8 and run it for a few days if you like.

**Comment by extremeshok** [comment]
@shawniverson @dandanio @Mrothyr @m0urs @msapiro @orlitzky @SlothOfAnarchy @jkellerer @RobbieTheK please can you test , all my tested have passed and this seems ready for release.

**Comment by dandanio** [comment]
Wait, there are some typos, give me a sec for a PR.

**Comment by extremeshok** [comment]
> Wait, there are some typos, give me a sec for a PR.

<===----- dyslexic...  http://geon.github.io/programming/2016/03/03/dsxyliea

**Comment by dandanio** [comment]
> <===----- dyslexic...

There is nothing wrong with that. Hold on, fixing typos and your Markdown... :) Give me some time. :)

**Comment by dandanio** [comment]
PR is in. :)

**Comment by jkellerer** [comment]
@extremeshok , the changes work for me. Thanks a lot for the updates.

**Comment by msapiro** [comment]
There's a lot of stuff that I don't exercise, but what I do seems good to me.

**Comment by shawniverson** [comment]
I am preparing this for test repositories on CentOS 7 and CentOS 8.  Should have some feedback shortly.

**Comment by shawniverson** [comment]
Initial testing is good on my end.

**Comment by extremeshok** [comment]
macOS / OSX is working on my mac, travis-ci appears to have issues with homebrew on their osx images

**Comment by extremeshok** [comment]
fixes #333

**Pull request merged**